examples/website/safegraph/app.js

```diff
@@ -79,14 +84,18 @@ function renderLayers(map, data) {
   const selectPOI = hex => {
     const [lat, lng] = cellToLatLng(hex);
     selectedPOICentroid = [lng, lat];
-    arcLayer.setProps({
-      data: data.filter(d => d.hex === hex)
-    });
+    deckOverlay.setProps({ 
+      layers: [
+        poiLayer, // TODO: This was written with an imperative pattern, which won't work for MapboxOverlay. poiLayer needs to be defined.
+        arcLayer.clone({
+          data: data.filter(d => d.hex === hex)
+        })
+      ]
+    })
```

Comment:
I'd like to hear some opinions on how to demonstrate this kind of partial update in an example. `MapboxLayer` supported it with a special `setProps` function added to each layer, but this is inconsistent with the majority of our examples and isn't supported when regular layers are in use, as is the case when using `MapboxOverlay`.

Reply:
The same behavior can be done using the `MaskExtension`: https://github.com/visgl/deck.gl/blob/master/test/apps/mask/app.jsx

If not using that, I think it is OK to recreate all the layers for the `setProps` call, deck.gl should do the diff internally and only update what is needed

Reply:
The common way of doing this is to change the signature to `renderLayers(deckOverlay, data, selectedPOI)` and call it from `onClick`.

Reply:
Thanks! Got it to work.
<img width="1404" alt="mapbox-v9" src="https://github.com/visgl/deck.gl/assets/2461547/cf9af9d9-6ebb-4141-8523-a00ae580e781">